### PR TITLE
feat(artifacthub): push artifacthub-repo.yml file to oci repository

### PR DESCRIPTION
**Pull request by asdfgugus**
issue reference: https://github.com/artifacthub/hub/issues/3919
docs: https://artifacthub.io/docs/topics/repositories/helm-charts/#oci-support

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -65,3 +65,17 @@ jobs:
...
           done
+      - name: Install oras
+        env:
+          ORAS_VERSION: 1.2.0
+        run: |
+          curl -LO "https://github.com/oras-project/oras/releases/download/v${VERSION}/oras_${VERSION}_linux_amd64.tar.gz"
+          mkdir -p oras-install/
+          tar -zxf oras_${VERSION}_*.tar.gz -C oras-install/
+          sudo mv oras-install/oras /usr/local/bin/
+          rm -rf oras_${VERSION}_*.tar.gz oras-install/
```
Comment (kranurag7) [inline .github/workflows/publish.yml:76]: We can also use setup-oras directly from upstream.
Ref: https://github.com/oras-project/setup-oras

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -65,3 +65,17 @@ jobs:
...
+      - name: Push artifacthub-repo.yml to GHCR
+        run: |
+          oras push registry/namespace/chart-name:artifacthub.io  \
```
Comment (kranurag7) [inline .github/workflows/publish.yml:79]:
```suggestion
          oras push ghcr.io/opencost/charts/opencost:artifacthub.io \
```

**Review by kranurag7 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (asdfgugus) [reply .github/workflows/publish.yml:76]: Thanks for your review! Silly mistakes... 
This is why you should not push late night...

**Review by kranurag7 — APPROVED** [review]
LGTM, thanks for the quick fixes @asdfgugus

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -65,3 +65,11 @@ jobs:
             helm push "${pkg}" oci://ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts |& tee .digest
             cosign sign $(cat .digest | awk -F "[, ]+" '/Pushed/{print $NF}')
           done
+      - uses: oras-project/setup-oras@v1
+        with:
+          oras-version: 1.2.0
```
Comment (kranurag7) [inline .github/workflows/publish.yml:70]:
```suggestion
          version: 1.2.0
```

I missed this in first go. This should be version as per the docs. 
Ref: https://github.com/oras-project/setup-oras?tab=readme-ov-file#advanced-usage

Night time for me as well :)

**Review by kranurag7 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 3 above)
Comment (asdfgugus) [reply .github/workflows/publish.yml]: This time it was autocomplete on my side...
A clear sign to turn that off ;)
Thanks again!

**Review by kranurag7 — APPROVED** [review]
Thanks again, LGTM.

**Review by toscott — APPROVED** [review]

**Pull request merged**